### PR TITLE
Clean couchforms views

### PR DESCRIPTION
**Pull request by dannyroberts**
Reopening most of https://github.com/dimagi/commcare-hq/pull/6879

This will opens the way for removing `couchforms/by_xmlns`

### CODE REVIEW

**Comment by dannyroberts** [comment]
@czue cc: @proteusvacuum @orangejenny

**Comment by proteusvacuum** [comment]
Conflict :crying_cat_face:

File: corehq/pillows/sofabed.py [hunk 1]
```diff
@@ -22,8 +23,12 @@ def process_sql(self, doc_dict, delete=False):
                 pass
             return
 
+        if doc_dict.get('xmlns') == DEVICE_LOG_XMLNS:
```
Comment (benrudolph) [inline corehq/pillows/sofabed.py]: fyi, you shouldn't have to worry about this case anymore since device logs are now being saved straight to SQL

Comment (benrudolph) [inline corehq/pillows/sofabed.py]: seems like you were witnessing the contrary though

Comment (benrudolph) [inline corehq/pillows/sofabed.py]: oh wait, if you ever reindex, you'll run into old docs that are device logs. :+1:

File: corehq/apps/hqadmin/dbaccessors.py [hunk 2]
```diff
@@ -17,6 +17,15 @@ def get_all_forms_in_all_domains():
...
+    """
+    Return number of non-error forms (but incl. logs) total across all domains
+    specifically as stored in couch.
+
+    (Can't rewrite to pull from ES or SQL; this function is used as a point
+    of comparison between row counts in other stores.)
+
+    """
+
     return XFormInstance.view('hqadmin/forms_over_time').one()['value']
```
Comment (snopoke) [inline corehq/apps/hqadmin/dbaccessors.py]: I don't think anyone ever really looks at those numbers so could probably kill that feature

Comment (dannyroberts) [inline corehq/apps/hqadmin/dbaccessors.py]: Do you mean we can kill this whole [db_comparisons](https://github.com/dimagi/commcare-hq/blob/01464841f387741fa8c6b6f58534087230618c90/corehq/apps/hqadmin/views.py#L354-L396) feature?

**Comment by dannyroberts** [comment]
:white_check_mark:

**Comment by dannyroberts** [comment]
(@snopoke I'll look for your answer, which could lead to a second PR where I kill that feature)

**Comment by snopoke** [comment]
@dannyroberts yup, kill that whole feature

File: corehq/apps/hqadmin/dbaccessors.py (hunk 2 above)
Comment (dannyroberts) [inline corehq/apps/hqadmin/dbaccessors.py]: ok that'll look like https://github.com/dimagi/commcare-hq/compare/clean-couchforms-views...kill-db_comparisons. It unfortunately doesn't really let us get rid of anything else like I was hoping, but still worth it if we're not using I guess

Comment (dannyroberts) [inline corehq/apps/hqadmin/dbaccessors.py]: (Not putting it in this PR just so I can throw it up on staging first and make sure it doesn't crash the status page.)

Comment (snopoke) [inline corehq/apps/hqadmin/dbaccessors.py]: :+1: :skull:

**Comment by dannyroberts** [comment]
@snopoke @czue any reason not to merge this in?

File: corehq/ex-submodules/couchforms/_design/util/dates.js [hunk 3]
```diff
@@ -1,21 +0,0 @@
-// parse a date in yyyy-mm-dd format
-function parse_date(date_string) {
```
Comment (czue) [inline corehq/ex-submodules/couchforms/_design/util/dates.js]: are you sure this doesn't trigger a reindex?

**Comment by czue** [comment]
:+1: no reason not to merge if you're confident that deleting something in _design/util doesn't cause a view rebuild. i'm not merging because i'm not sure

**Comment by snopoke** [comment]
I'm sure changing code in utils will cause a reindex since couchdbkit just
compiles it all into one js file.

**Comment by czue** [comment]
thanks, added "reindex" label

**Comment by dannyroberts** [comment]
There's no reindex because dates.js only get included if you explicitly include it in a view with

```
// !code util/dates.js
```

and nothing in `couchforms/_design` references dates.js:

```bash
$ git checkout master
$ ag dates.js corehq/ex-submodules/couchforms
$
```

**Comment by czue** [comment]
ah, nice. can you confirm you ran a local reindex and it worked ok?

**Comment by dannyroberts** [comment]
Yup, confirmed

**Comment by dannyroberts** [comment]
:haircut:

**Comment by snopoke** [comment]
🐀

File: corehq/apps/hqadmin/dbaccessors.py (hunk 2 above)
Comment (dannyroberts) [inline corehq/apps/hqadmin/dbaccessors.py]: https://github.com/dimagi/commcare-hq/pull/9565

**Pull request merged**